File: backend/src/main/java/com/bsuir/course/entity/Room.java
package com.bsuir.course.entity;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.Objects;

@Entity
@Table(name = "rooms", schema = "hostel")
public class Room {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private long id;

    @NotNull
    @Column(name = "number")
    private int number;


    @NotNull
    @Column(name = "price")
    private int price;


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Room room = (Room) o;
        return id == room.id &&
                number == room.number &&
                price == room.price;
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, number, price);
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }


    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public int getNumber() {
        return number;
    }

    public void setNumber(int number) {
        this.number = number;
    }
}

File: frontend/my-app/src/app/model/room.ts
export class Room{
  id:number;
  number:number;
  price:number;
  numberOfClients:number;
  static cloneBaseRoom(room:Room):Room{
    let clonedRoom:Room = new Room();
    clonedRoom.id=room.id;
    clonedRoom.number=room.number;
    clonedRoom.price=room.price;
    clonedRoom.numberOfClients=room.numberOfClients;
    return clonedRoom;
  }
}
